notes/stv/20200507-hadoop-k8s.txt

```diff
@@ -0,0 +1,422 @@
+	    HADOOP_CONF_DIR=/usr/local/hadoop/etc/hadoop \
+	    YARN_CONF_DIR=/usr/local/hadoop/etc/hadoop \
+	    PATH=${PATH}:/usr/local/hadoop/bin
+
+	RUN \
+	  cd /usr/local && ln -s ./hadoop-${HADOOP_VERSION} hadoop && \
+	  rm -f ${HADOOP_PREFIX}/logs/*
+
+	WORKDIR $HADOOP_PREFIX
+
+	# Hdfs ports
+	EXPOSE 50010 50020 50070 50075 50090 8020 9000
+	# Mapred ports
+	EXPOSE 19888
+	#Yarn ports
+	EXPOSE 8030 8031 8032 8033 8040 8042 8088
+	#Other ports
+	EXPOSE 49707 2122
+
+
+
+# ----------------------------------------------------------
+# Checking what was deployed
+# ----------------------------------------------------------
+
+
+
+# Get Pods
+# -------------
+kubectl get pods -n default
+NAME                            READY   STATUS    RESTARTS   AGE
+hadoop-hadoop-hdfs-dn-0         1/1     Running   2          128m
+hadoop-hadoop-hdfs-nn-0         1/1     Running   0          128m
+hadoop-hadoop-yarn-nm-0         1/1     Running   0          128m
+hadoop-hadoop-yarn-nm-1         1/1     Running   0          128m
+hadoop-hadoop-yarn-rm-0         1/1     Running   0          128m
+spark-master-controller-4jnbd   1/1     Running   0          9d
+spark-worker-controller-4jvh5   1/1     Running   0          9d
+spark-worker-controller-rw4pq   1/1     Running   0          9d
+
+
+# There are 5 Hadoop pods:
+1 data node
+3 name nodes
+1 resource manager node
+
+# I assume 1 datanode cause "replicas" param defaults to one?
+
+
+
+# Get Services
+# -------------
+kubectl get services     
+NAME                    TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                      AGE
+hadoop-hadoop-hdfs-dn   ClusterIP   None             <none>        9000/TCP,50075/TCP           132m
+hadoop-hadoop-hdfs-nn   ClusterIP   None             <none>        9000/TCP,50070/TCP           132m
+hadoop-hadoop-yarn-nm   ClusterIP   None             <none>        8088/TCP,8082/TCP,8042/TCP   132m
+hadoop-hadoop-yarn-rm   ClusterIP   None             <none>        8088/TCP                     132m
+hadoop-hadoop-yarn-ui   ClusterIP   10.254.142.3     <none>        8088/TCP                     132m
+kubernetes              ClusterIP   10.254.0.1       <none>        443/TCP                      10d
+spark-master            ClusterIP   10.254.252.249   <none>        7077/TCP,8080/TCP            9d
+spark-master-headless   ClusterIP   None             <none>        <none>                       9d
+
+
+# Get Persistent Storage
+# ----------------------
+kubectl get pv --sort-by=.spec.capacity.storage
+NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                           STORAGECLASS   REASON   AGE
+pvc-e62eef27-6a06-447d-867f-13407c6e6fa7   50Gi       RWO            Delete           Bound    default/hadoop-hadoop-hdfs-nn   standard                139m
+pvc-b995c154-ddd0-4fdb-a48a-0a25b75a5f06   200Gi      RWO            Delete           Bound    default/hadoop-hadoop-hdfs-dn   standard                140m
```

Comment:
Lots of questions
* How does K8s know which Openstack volumes to use ?
* If 100Gi was added to 4 worker nodes, why do we get 200Gi on hdfs-dn ?
* What defines the mapping between Openstack volumes and Kubernetes storage ?